source/solvers/navier_stokes_base.cc

```diff
@@ -1312,6 +1314,101 @@ NavierStokesBase<dim, VectorType, DofsType>::postprocess_fd(bool firstIter)
     }
 
 
+  // Calculate momentum drop between two boundaries
+  if (this->simulation_parameters.post_processing.calculate_momentum_drop)
+    {
+      double momentum_drop = calculate_momentum_drop(
+        this->dof_handler,
+        this->mapping,
+        this->evaluation_point,
+        *this->face_quadrature,
+        this->simulation_parameters.post_processing.inlet_boundary_id,
+        this->simulation_parameters.post_processing.outlet_boundary_id,
+        simulation_parameters.physical_properties_manager);
+      this->momentum_drop_table.add_value(
+        "time", simulation_control->get_current_time());
+      this->momentum_drop_table.add_value("momentum-drop", momentum_drop);
+      if (this->simulation_parameters.post_processing.verbosity ==
+          Parameters::Verbosity::verbose)
+        {
+          this->pcout << "Momentum drop: "
+                      << std::setprecision(
+                           simulation_control->get_log_precision())
+                      << this->simulation_parameters.physical_properties_manager
+                             .get_density_scale() *
+                           momentum_drop
+                      << " kg/(m*s^2)." << std::endl;
+        }
+
+      // Output momentum drop to a text file from processor 0
+      if ((simulation_control->get_step_number() %
+             this->simulation_parameters.post_processing.output_frequency ==
+           0) &&
+          this->this_mpi_process == 0)
+        {
+          std::string filename =
+            simulation_parameters.simulation_control.output_folder +
+            simulation_parameters.post_processing.momentum_drop_output_name +
+            ".dat";
+          std::ofstream output(filename.c_str());
+          momentum_drop_table.set_precision("time", 12);
+          momentum_drop_table.set_precision("momentum-drop", 12);
```

Comment:
Does this really require an independent file or it could be in the same file as the pressure drop? Otherwise I feel we are generating a ton of file for no reason. E.g. when you calculate pressure drop we could also calculate the total momentum drop and put i in the same file